File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] ans = new int[nums.length];

        int p1 = 0;
        int p2 = nums.length - 1;
        int p3 = ans.length - 1;

        while (p3 > -1) {
            int val1 = nums[p1] * nums[p1];
            int val2 = nums[p2] * nums[p2];

            if (val1 > val2) {
                ans[p3] = val1;
                p1++;
            } else {
                ans[p3] = val2;
                p2--;
            }

            p3--;
        }

        return ans;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function (nums) {
    let ans = [];
    ans.length = nums.length

    let p1 = 0;
    let p2 = nums.length - 1;

    for (let i = nums.length - 1; i >= 0; i--) {
        let val1 = nums[p1] * nums[p1];
        let val2 = nums[p2] * nums[p2];

        if (val1 < val2) {
            ans[i] = val2;
            p2--;
        } else {
            ans[i] = val1;
            p1++;
        }
    }

    return ans;
};